### PR TITLE
switch CI jobs between windows and linux for example execution

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Example execution on linux/vulkan on CI is segfaulting for unclear reasons
- This makes a lot of noise on PRs

## Solution

- Switch example execution on Linux to validation jobs (on PR merged). It will still crash but not block merging, and we'll know when it's fixed
- Switch example execution on Windows to CI jobs (on PR push). It's a bit longer than on Linux but provides a useful status
- Disable job commenting on PR with job execution to reduce noise

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
I'm on board with the swap, but I'd prefer not to have commented out code. That's what version control is for! Don't block on this though.

**Comment by cart** [comment]
This does seem to increase the run time of the job pretty significantly. The first run finished in 26 minutes, vs the 11-15 minutes on linux. If its 26 minutes every time, it will add an extra 10-ish minutes to every build (which used to take 15 ish minutes with the linux variant).

I trigger a second run for another datapoint.

**Comment by cart** [comment]
Second run was 15 minutes. Nice!

**Review by cart — APPROVED** [review]
Also on board, but I agree that we should just remove the commented out section

File: .github/workflows/ci-comment-failures.yml [hunk 1]
```diff
@@ -14,54 +14,55 @@ on:
...
+  #         script: |
+  #           var fs = require('fs');
+  #           var issue_number = Number(fs.readFileSync('./NR'));
+  #           var last_example_run = fs.readFileSync('./last_example_run');
+  #           await github.rest.issues.createComment({
+  #             owner: context.repo.owner,
+  #             repo: context.repo.repo,
+  #             issue_number: issue_number,
+  #             body: 'Example `' + last_example_run + '` failed to run, please try running it locally and check the result.'
+  #           });
```
Comment (alice-i-cecile) [inline .github/workflows/ci-comment-failures.yml:65]:
```suggestion
```

File: .github/workflows/validation-jobs.yml [hunk 2]
```diff
@@ -78,23 +86,48 @@ jobs:
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: screenshots.zip
+          path: screenshots.zip
+      # - name: Save PR number
+      #   if: ${{ failure() && github.event_name == 'pull_request' }}
+      #   run: |
+      #     mkdir -p ./example-run
+      #     echo ${{ github.event.number }} > ./example-run/NR
+      #     mv last_example_run ./example-run/
```
Comment (alice-i-cecile) [inline .github/workflows/validation-jobs.yml:125]:
```suggestion
```

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**